### DISCUSSION TITLE
How do I generate new keys and pem files like in `examples/data/tls`?

### DISCUSSION THREAD

**Discussion by sadeli413** [hyperium/tonic#1053] [Q&A]
I want to follow the example shown in `examples/src/tls_client_auth` but I don't know how to create my own certs like in `examples/data/tls`.

So far, I have tried using certificates with [mkcert](https://github.com/FiloSottile/mkcert) and using openssl like in [this example](https://github.com/camelop/rust-mtls-example).

But these do not generate similar certs, and the tls_client_auth example panics unless I use the provided certs in the tonic repo.

**Comment by LucioFranco** [comment]
I actually did use `mkcert` to generate those certs but I don't remember what commands I used at this point.
